xmppserver/src/main/java/org/jivesoftware/openfire/muc/MUCRoom.java

```diff
@@ -1432,6 +1434,9 @@ public void destroyRoom(JID alternateJID, String password, String reason) {
         MUCPersistenceManager.deleteFromDB(this);
         // Remove the history of the room from memory (preventing it to pop up in a new room by the same name).
         roomHistory.purge();
+        // If we are not preserving room history on deletion, fire event to clear chat room history
+        if(!preserveHistOnRoomDeletion)
```

Comment:
I don't care much about people using different coding styles, but not using curly braces on single-line if-statements is an accident waiting to happen. Please use something like this instead:

```
if(!preserveHistOnRoomDeletion) {
    MUCEventDispatcher.roomClearChatHistory(getSelfRepresentation().getOccupantJID());
}
```

Reply:
Also here `getJID()` rather than the occupant JID from the self-representation is probably preferred.

Reply:
Done.